orion/pkg/repos/announce_repo_test.go

```diff
@@ -0,0 +1,165 @@
+package repos_test
+
+import (
+	"database/sql"
+	sqlmock "github.com/DATA-DOG/go-sqlmock"
+	"github.com/ahsu1230/mathnavigatorSite/orion/pkg/domains"
+	"github.com/ahsu1230/mathnavigatorSite/orion/pkg/repos"
+	"reflect"
+	"testing"
+	"time"
+)
+
+func initAnnounceTest(t *testing.T) (*sql.DB, sqlmock.Sqlmock, repos.AnnounceRepoInterface) {
+	db, mock, err := sqlmock.New()
+	if err != nil {
+		t.Fatalf("an error '%s' was not expected when opening a stub database connection", err)
+	}
+	repo := repos.CreateTestAnnounceRepo(db)
+	return db, mock, repo
+}
+
+//
+// Test Select All
+//
+func TestSelectAllAnnouncements(t *testing.T) {
+	db, mock, repo := initAnnounceTest(t)
+	defer db.Close()
+
+	// Mock DB statements and execute
+	now := time.Now().UTC()
+	rows := sqlmock.NewRows([]string{"Id", "CreatedAt", "UpdatedAt", "DeletedAt", "PostedAt", "Author", "Message"}).AddRow(1, now, now, sql.NullTime{}, now, "Author Name", "Valid Message")
+	mock.ExpectPrepare("^SELECT (.+) FROM announcements").ExpectQuery().WillReturnRows(rows)
```

Comment:
apply comment from below to statements 31 & 32.

For the rows part, maybe it could look something like this:
```
rows := sqlmock.NewRows([]string{
    "Id",
    "CreatedAt",
    ....
})
```

Reply:
The format in `program_repo_test.go` should also follow this format too